vllm/lora/models.py

```diff
@@ -190,67 +191,122 @@ def from_local_checkpoint(
+                    )
+
+                return cls.from_lora_tensors(
+                    lora_model_id=get_lora_id()
+                    if lora_model_id is None else lora_model_id,
+                    tensors=tensors,
+                    peft_helper=peft_helper,
+                    device=device,
+                    dtype=dtype,
+                    embeddings=embeddings,
+                    target_embedding_padding=target_embedding_padding,
+                    embedding_modules=embedding_modules,
+                    embedding_padding_modules=embedding_padding_modules,
+                    weights_mapper=weights_mapper)
+            except LoRASourceError as e:
+                # Preserve the original error details
+                if hasattr(e, 'original_error') and e.original_error:
+                    logger.error("LoRA loading error: %s, caused by: %s",
+                                 str(e), str(e.original_error))
+                else:
+                    logger.error("LoRA loading error: %s", str(e))
+                raise ValueError(
+                    f"Failed to load LoRA adapter: {str(e)}") from e
+            except Exception as e:
+                logger.error("Unexpected error loading LoRA: %s", str(e))
+                raise ValueError(
+                    f"Failed to load LoRA adapter: {str(e)}") from e
+
+        # Existing local path logic
+        lora_dir = lora_dir_or_source
         lora_config_path = os.path.join(lora_dir, "adapter_config.json")
         lora_tensor_path = os.path.join(lora_dir, "adapter_model.safetensors")
         lora_bin_file_path = os.path.join(lora_dir, "adapter_model.bin")
         new_embeddings_tensor_path = os.path.join(
             lora_dir, "new_embeddings.safetensors")
         new_embeddings_bin_file_path = os.path.join(lora_dir,
                                                     "new_embeddings.bin")
+
         with open(lora_config_path) as f:
             config = json.load(f)
 
         config["vllm_max_position_embeddings"] = max_position_embeddings
         peft_helper = PEFTHelper.from_dict(config)
-        unexpected_modules: List[Union[list[str], str]]
+
         if os.path.isfile(lora_tensor_path):
-            tensors: Dict[str, torch.Tensor] = {}
+            model_tensors: Dict[str, torch.Tensor] = {}
             # Find unexpected modules.
             # Use safetensor key as a source of truth to find expected modules.
             # in peft if you have target_modules A, B, C and C does not exist
-            # in the model it won’t error and model will be trained with A, B
-            # loraified. C won’t exist in the safetensor but it will exist in
+            # in the model it won't error and model will be trained with A, B
+            # loraified. C won't exist in the safetensor but it will exist in
             # the target_modules of the adapter_config.json.
-            unexpected_modules = []
+            found_modules = []
```

Comment:
I think the name `unexpected_modules` conveyed the meaning that these are the modules that we did not expect to find. 
The name `found_modules` I think is a bit misleading, since it's not all the modules that we found. It's actually `found_modules - expected_lora_modules`